packages/date-picker/src/basic/time-spinner.vue

```diff
@@ -60,7 +60,7 @@
             class="el-time-spinner__item"
             :class="{ 'active': hour === hours, 'disabled': hoursList[hour] }"
             v-for="hour in arrowHourList">
-            {{ hour === undefined ? '' : ('0' + hour).slice(-2) }}
+            {{ ('0' + (amPmMode ? (hour % 12 || 12) : hour )).slice(-2) }}{{amPm(hour)}}
```

Comment:
You should handle `hour === undefined` here

Reply:
Sorry for it. Revised.